src/clients/c++/request.h

```diff
@@ -277,6 +277,11 @@ class InferContext {
     /// -1.
     virtual const DimsList& Dims() const = 0;
 
+    virtual bool IsSharedMemory() const = 0;
```

Comment:
The Input class provides methods to query the model configuration. Not methods to learn about setting you've made to the input. So these IsShared... GetShared... methods should be removed.